crates/xilem_html/src/either.rs

```diff
@@ -0,0 +1,225 @@
+                match (prev, self) {
+                    $(
+                        // Variant is the same as before
+                        ($ident::$vars(prev_view), $ident::$vars(view_sequence)) => {
+                            let $ident::$vars(state) = state else {
+                                throw_str(concat!(
+                                    "invalid state/view_sequence in ",
+                                    stringify!($ident),
+                                    " (unreachable)",
+                                ));
+                            };
+                            view_sequence.rebuild(cx, prev_view, state, element)
+                        }
+                        // Variant has changed
+                        (_, $ident::$vars(view_sequence)) => {
+                            let new_state =
+                                element.as_vec(|elements| view_sequence.build(cx, elements));
+                            *state = $ident::$vars(new_state);
+                            ChangeFlags::STRUCTURE
+                        }
+                    )+
+                }
+            }
+
+            fn message(
+                &self,
+                id_path: &[xilem_core::Id],
+                state: &mut Self::State,
+                message: Box<dyn std::any::Any>,
+                app_state: &mut VT,
+            ) -> xilem_core::MessageResult<VA> {
+                match self {
+                    $(
+                        $ident::$vars(view_sequence) => {
+                            let $ident::$vars(state) = state else {
+                                throw_str(concat!(
+                                    "invalid state/view_sequence in ",
+                                    stringify!($ident),
+                                    " (unreachable)",
+                                ));
+                            };
+                            view_sequence.message(id_path, state, message, app_state)
+                        }
+                    )+
+                }
+            }
+
+            fn count(&self, state: &Self::State) -> usize {
+                match self {
+                    $(
+                        $ident::$vars(view_sequence) => {
+                            let $ident::$vars(state) = state else {
+                                throw_str(concat!(
+                                    "invalid state/view_sequence in ",
+                                    stringify!($ident),
+                                    " (unreachable)",
+                                ));
+                            };
+                            view_sequence.count(state)
+                        }
+                    )+
+                }
+            }
+        }
+
+    };
+}
+
+either! {
+    /// This view container can switch between two views.
+    Either { A, B }
+}
+either! {
+    /// This view container can switch between three views.
+    Either3 { A, B, C }
```

Comment:
Multiples > 2 are usually called `OneOf` so I would suggest using this term instead

Reply:
I've actually seen `EitherN` more times in Rust context, but I agree that `OneOf` is a bit more natural. Wdyt should happen to `Either` in that case? Should it stay as-is, become `OneOf`, or maybe `OneOf2`?

Reply:
I think `Either` is fine for two elements, feels more natural than `OneOf2` or `OneOf` IMHO (and is used as well more often in functional programming)

Reply:
Hm, then I'm also wondering if it still makes sense to have `Either`s variants keep the `Left` / `Right` variant names.

But thinking more about it, I think I'd really prefer `Either` to be `OneOf2` for consistency.

Maybe somebody else wants to chime in as well to give their opinion on the color of the bikeshed? 😄